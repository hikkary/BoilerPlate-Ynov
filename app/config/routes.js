import { createSwitchNavigator, createStackNavigator } from 'react-navigation'

import AuthLoading from '../screens/authLoading'
import Home from '../screens/home'
import Login from '../screens/login'
import Options from '../screens/options'
import Themes from '../screens/themes'

const optionsGeneral = {
  mode: 'modal',
  headerMode: 'none'
}

const LoggedUser = createStackNavigator({
  Home: {
    screen: Home,
    navigationOptions: {
      header: () => null
    }
  },
  Options: {
    screen: Options,
    navigationOptions: {
      headerTitle: 'Options'
    }
  },
  Themes: {
    screen: Themes,
    navigationOptions: {
      headerTitle: 'Themes'
    }
  }
})

const DisconnectedUser = createStackNavigator({
  Login: {
    screen: Login,
    navigationOptions: {
      header: () => null
    }
  }
})

export default createSwitchNavigator(
  {
    AuthLoading: { screen: AuthLoading },
    Auth: {
      screen: DisconnectedUser
    },
    Home: {
      screen: LoggedUser
    }
  },
  optionsGeneral
)
