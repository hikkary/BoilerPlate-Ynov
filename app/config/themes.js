export const theme = {
  color: {
    primary: '#7ba9cc',
    secondary: '#6C94B2'
  }
}
