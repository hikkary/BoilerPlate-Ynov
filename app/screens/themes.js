import { Text } from 'react-native'
import React, { Component } from 'react'
import styled from 'styled-components'

const ThemesContainer = styled.View`
    align-Items: 'center',
    background-color: '#F5FCFF'
    flex: 1,
    justify-content: 'center',
`

export default class Themes extends Component {
  render() {
    return (
      <ThemesContainer>
        <Text>Couleur One</Text>
        <Text>Couleur Two</Text>
        <Text>Couleur Three</Text>
      </ThemesContainer>
    )
  }
}
