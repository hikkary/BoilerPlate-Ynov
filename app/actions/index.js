import * as counter from './counter'
import * as themes from './themes'

export default { counter, themes }
