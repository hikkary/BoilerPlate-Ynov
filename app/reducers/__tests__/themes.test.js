import reducer from '../themes'

import { themeBlue, themeRed, themeGreen } from '../../config/themes'
import { changeTheme } from '../../actions/themes'

const initialState = {
  currentTheme: themeBlue
}

// Test Driven Devlopment

it('sets initial state', () => {
  const expected = { ...initialState }
  const actual = reducer(undefined, {})
  expect(actual).toEqual(expected)
})

it('set themes on red ', () => {
  const expected = {
    ...initialState,
    currentTheme: {
      ...themeRed
    }
  }

  const actual = reducer(undefined, changeTheme(themeRed))
  expect(actual).toEqual(expected)
})

it('set themes on green ', () => {
  const expected = {
    ...initialState,
    currentTheme: {
      ...themeGreen
    }
  }

  const actual = reducer(undefined, changeTheme(themeGreen))
  expect(actual).toEqual(expected)
})

it('set themes on blue ', () => {
  const expected = {
    ...initialState,
    currentTheme: {
      ...themeBlue
    }
  }

  const actual = reducer(undefined, changeTheme(themeBlue))
  expect(actual).toEqual(expected)
})

// MATCH SNAPSHOT

it('sets initial state Snapshot', () => {
  expect(reducer(undefined, {})).toMatchSnapshot()
})

it('set themes on red Snapshot ', () => {
  expect(reducer(undefined, changeTheme(themeRed))).toMatchSnapshot()
})

it('set themes on green Snapshot', () => {
  expect(reducer(undefined, changeTheme(themeGreen))).toMatchSnapshot()
})

it('set themes on blue Snapshot', () => {
  expect(reducer(undefined, changeTheme(themeBlue))).toMatchSnapshot()
})
