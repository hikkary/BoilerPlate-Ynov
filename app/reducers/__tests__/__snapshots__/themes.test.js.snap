// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set themes on blue Snapshot 1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "primary": "#7ba9cc",
      "secondary": "#6C94B2",
    },
  },
}
`;

exports[`set themes on green Snapshot 1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "primary": "#95D4AB",
      "secondary": "#44875B",
    },
  },
}
`;

exports[`set themes on red Snapshot  1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "primary": "#D49795",
      "secondary": "#875352",
    },
  },
}
`;

exports[`sets initial state Snapshot 1`] = `
Object {
  "currentTheme": Object {
    "color": Object {
      "primary": "#7ba9cc",
      "secondary": "#6C94B2",
    },
  },
}
`;
